src/loaders/workspace_uploader/workspace_uploader.py

```diff
@@ -395,52 +372,88 @@ def _upload_assembly_files_in_parallel(
         workspace_id: target workspace id
         upload_dir: a directory in collectionssource that creates new directories linking to sourcedata
         wait_to_upload_assemblies: a dictionary that maps assembly file name to assembly directory
+        batch_size: a number of files to upload per batch
 
     Returns:
-        a list of assembly names that failed to upload
+        number of assembly files have been sucessfully uploaded from wait_to_upload_assemblies
     """
     assembly_files_len = len(wait_to_upload_assemblies)
-    print(f"Start uploading {assembly_files_len} assembly files with {conf.workers} workers\n")
+    print(f"Start uploading {assembly_files_len} assembly files\n")
 
-    # Put the assembly files in the input_queue
-    counter = 1
-    for assembly_name, host_assembly_dir in wait_to_upload_assemblies.items():
+    uploaded_count = 0
+    for assembly_tuples in _gen(wait_to_upload_assemblies, batch_size):
 
-        container_internal_assembly_path = os.path.join(
-            JOB_DIR_IN_ASSEMBLYUTIL_CONTAINER, assembly_name
-        )
-        conf.input_queue.put(
-            (
+        try:
+            batch_upas = _upload_assemblies_to_workspace(conf, workspace_id, assembly_tuples)
+        except Exception as e:
+            failed_objects_names = [assembly_tuple.assembly_name for assembly_tuple in assembly_tuples]
+            assembly_objects_in_ws = loader_helper.list_objects(
+                workspace_id,
+                conf,
+                loader_common_names.OBJECTS_NAME_ASSEMBLY,
+            )
+            assembly_names_in_ws = [object_info[1] for object_info in assembly_objects_in_ws]
+            objects_names_to_delete_from_ws = set(assembly_names_in_ws) & set(failed_objects_names)
+            if objects_names_to_delete_from_ws:
```

Comment:
Why delete good objects? I can see retrying or reporting the failed uploads but deleting successful uploads seems counter productive

Reply:
Updated!

Reply:
:+1: